hydra-cluster/src/Hydra/Cluster/Faucet.hs

```diff
@@ -127,18 +127,21 @@ returnFundsToFaucet tracer node@RunningNode{networkId, nodeSocket} sender = do
   utxo <- queryUTxOFor networkId nodeSocket QueryTip senderVk
 
   retryOnExceptions tracer $ do
-    let allLovelace = selectLovelace $ balance @Tx utxo
+    let utxoValue = balance @Tx utxo
+    let allLovelace = selectLovelace utxoValue
+    -- select tokens other than ADA here so we can burn it afterwards
```

Comment:
We can't just burn tokens. Sending them back to the faucet is our only option. So this calculation would be more clear if we would just use `utxoValue` and do a `<> negateValue (lovelaceToValue fee)` or so